--- dwm.c.orig	2013-10-04 18:19:22.858118000 -0700
+++ b/dwm-6.0/dwm.c	2013-10-04 18:17:23.004118000 -0700
@@ -147,6 +147,7 @@ typedef struct {
 struct Monitor {
 	char ltsymbol[16];
 	int num;
+	int status;
 	int by;               /* bar geometry */
 	int mx, my, mw, mh;   /* screen size */
 	int wx, wy, ww, wh;   /* window area  */
@@ -750,6 +751,7 @@ createmon(void) {
 	if(!(m->lts = calloc(numtags, sizeof(Layout *))))
 		die("fatal: could not malloc() %u bytes\n", sizeof(Layout *) * numtags);
 	m->tagset[0] = m->tagset[1] = 1;
+	m->status = 0;
 	m->mfacts[0] = mfact;
 	m->nmasters[0] = nmaster;
 	m->lts[0] = &layouts[0];
@@ -851,7 +853,7 @@ drawbar(Monitor *m) {
 	drawtext(m->ltsymbol, dc.colors[0], False);
 	dc.x += dc.w;
 	x = dc.x;
-	if(m == selmon) { /* status is only drawn on selected monitor */
+	if(m->status) { /* status is only drawn on `statusmon` */
 		dc.w = textnw(stext, strlen(stext)); // no padding
 		dc.x = m->ww - dc.w;
 		if(showsystray && m == selmon) {
@@ -2131,6 +2133,7 @@ updategeom(void) {
 					m->my = m->wy = unique[i].y_org;
 					m->mw = m->ww = unique[i].width;
 					m->mh = m->wh = unique[i].height;
+					m->status = (i == statusmon ? 1 : 0);
 					updatebarpos(m);
 				}
 		}
@@ -2159,6 +2162,7 @@ updategeom(void) {
 	{
 		if(!mons)
 			mons = createmon();
+		mons->status = 1;
 		if(mons->mw != sw || mons->mh != sh) {
 			dirty = True;
 			mons->mw = mons->ww = sw;
@@ -2248,9 +2252,11 @@ updatetitle(Client *c) {
 
 void
 updatestatus(void) {
+	Monitor* m;
 	if(!gettextprop(root, XA_WM_NAME, stext, sizeof(stext)))
 		strcpy(stext, "dwm-"VERSION);
-	drawbar(selmon);
+	for(m = mons; m; m = m->next)
+		drawbar(m);
 }
 
 void
